tests/sqlparser_common.rs

```diff
@@ -11387,6 +11387,372 @@ fn test_try_convert() {
+                        data_type: Some(DataType::Custom(
+                            ObjectName(vec![Ident {
+                                value: "XML".into(),
+                                quote_style: None
+                            }]),
+                            vec![]
+                        )),
+                        charset: None,
+                        target_before_value: true,
+                        styles: vec![]
+                    }
+                    .into(),
+                    right: Function {
+                        name: ObjectName(vec![Ident {
+                            value: "value".into(),
+                            quote_style: None
+                        }]),
+                        parameters: FunctionArguments::None,
+                        args: FunctionArguments::List(FunctionArgumentList {
+                            duplicate_treatment: None,
+                            args: vec![
+                                FunctionArg::Unnamed(FunctionArgExpr::Expr(Expr::Value(
+                                    Value::SingleQuotedString(".".into())
+                                ))),
+                                FunctionArg::Unnamed(FunctionArgExpr::Expr(Expr::Value(
+                                    Value::SingleQuotedString("NVARCHAR(MAX)".into())
+                                )))
+                            ],
+                            clauses: vec![]
+                        }),
+                        filter: None,
+                        null_treatment: None,
+                        over: None,
+                        within_group: vec![]
+                    }
+                })
+                .into(),
+                right: Function {
+                    name: ObjectName(vec![Ident {
+                        value: "value".into(),
+                        quote_style: None
+                    }]),
+                    parameters: FunctionArguments::None,
+                    args: FunctionArguments::List(FunctionArgumentList {
+                        duplicate_treatment: None,
+                        args: vec![
+                            FunctionArg::Unnamed(FunctionArgExpr::Expr(Expr::Value(
+                                Value::SingleQuotedString(".".into())
+                            ))),
+                            FunctionArg::Unnamed(FunctionArgExpr::Expr(Expr::Value(
+                                Value::SingleQuotedString("NVARCHAR(MAX)".into())
+                            )))
+                        ],
+                        clauses: vec![]
+                    }),
+                    filter: None,
+                    null_treatment: None,
+                    over: None,
+                    within_group: vec![]
+                }
+            }),
+            alias: Ident {
+                value: "T".into(),
+                quote_style: None
+            }
+        }],
+        sql.projection
+    );
```

Comment:
I find these tests hard to follow. And they will need to be updated even when the ast changes in places completely unrelated to Expr::CompositeFunction.

Maybe you could just test the new syntax introduced here with small precise asserts that test just a single thing ? Like:

```rs
assert!eq!(verified_expr("convert(XML, '<x/>').query('x')", some_small_ast);
```

If you want to ensure some more complex queries always parse, you could add some verified_only_select, without adding an assert on the result.

This should make the tests easier to understand, this pr easier to review, and reduce the burden of maintaining these tests in the future.